src/basilisp/contrib/nrepl_server.lpy

```diff
@@ -0,0 +1,478 @@
+(defmacro ^:private error [& values]
+  ;; we want error to always output to `sys/stderr` unfiltered.
+  `(.write sys/stderr (str ~(keyword (str *ns*) "error") " " (str/join " " [~@values]) \newline)))
+
+(definterface IReadWrite
+  ;; Pythonic interface for creating `sys/stdout` like File objects.
+  (read [])
+  (write [value]))
+
+(deftype StreamOutFn [out-fn]
+  ;; A type to use as replacement binding for writing to `sys/stdout`
+  ;; stream, so that the output ``value`` is passed to ``out-fn``
+  ;; instead.
+  IReadWrite
+  (read [_this]
+    (throw (python/RuntimeError (str ::logic-error " shouldn't be possible to get here..."))))
+  (write [_self value]
+    (out-fn value)))
+
+(defn- response-for-mw [handler]
+  (fn [{:keys [id session] :as request} response]
+    (let [response (cond-> (assoc response
+                                  "id" id)
+                     session (assoc "session" session))]
+      (handler request response))))
+
+(defn- coerce-request-mw [handler]
+  (fn [request send-fn]
+    (handler (update request :op keyword) send-fn)))
+
+(defn- log-request-mw [handler]
+  (fn [request send-fn]
+    (info :request (dissoc request :client*))
+    (handler request send-fn)))
+
+(defn- log-response-mw [handler]
+  (fn [request response]
+    (info :response response)
+    (handler request response)))
+
+(declare ops)
+
+(defn- handle-describe [request send-fn]
+  (send-fn request
+           {"versions" {"basilisp" (let [version (py->lisp ver/VERSION)]
+                                     (assoc (zipmap ["major" "minor" "incremental"]
+                                                    version)
+                                            "version-string" (str/join "." version)))
+                        "python" (let [version (.__getitem__ (.split sys/version " ") 0)]
+                                   (assoc (zipmap ["major" "minor" "incremental"]
+                                                  (py->lisp (.split version ".")))
+                                          "version-string" (py->lisp sys/version)))}
+            "ops" (zipmap (map name (keys ops)) (repeat {}))
+            "status" ["done"]}))
+
+(defn- format-value [_nrepl-pprint _pprint-options value]
+  (pr-str value))
+
+(defn- send-value [request send-fn v]
+  (let [{:keys [client*]} request
+        {:keys [*2 *3 *e]} @client*
+        [v opts] v
+        ns (:ns opts)]
+    (swap! client* assoc :*1 v :*2 *1 :*3 *2)
+    (let [v (format-value (:nrepl.middleware.print/print request)
+                          (:nrepl.middleware.print/options request)
+                          v)]
+      (send-fn request {"value" (str v)
+                        "ns" (str ns)}))))
+
+(defn- handle-error [send-fn request e]
+  (let [{:keys [client* ns]} request
+        data (ex-data e)
+        message (or (:message data) (str e))]
+    (swap! client* assoc :*e e)
+    (send-fn request {"err" (str message)})
+    (send-fn request {"ex" (traceback/format_exc)
```

Comment:
```suggestion
    (send-fn request {"ex" (traceback/format-exc)
```

Think we can use the kebab case here, right?

Reply:
I didn't know this was possible, updated.